### ISSUE TITLE
False positives on Coffee cup images

### ISSUE THREAD

**Issue by adnankhan37** [infinitered/nsfwjs#584]
Hello,

We have been using Inception v3 for NSFW detection and we are getting false positives on coffee cup images.

![image](https://user-images.githubusercontent.com/61727245/156025611-008b5ad5-ba01-45f9-a536-0df2320dbcd0.png)
![image](https://user-images.githubusercontent.com/61727245/156025801-008052bc-f0b3-441f-bb26-06eb4edf1d3a.png)
![image](https://user-images.githubusercontent.com/61727245/156025868-e11d51f4-a119-4431-825c-f1ffd71cc5be.png)
![image](https://user-images.githubusercontent.com/61727245/156025927-bcf6db7a-8458-4c81-b1d8-79eb8959821f.png)
![image](https://user-images.githubusercontent.com/61727245/156026050-e66fc7a4-4b7e-4efc-a612-c7f4e3710bdb.png)

Any suggestion on how can we tackle this?

**Comment by GantMan** [comment]
WOW!  This is a great find.  Which model are you using?  The default?  Does this happen on ALL the models in the dropdown?

**Comment by adnankhan37** [comment]
I am using Inception v3. Tried the mobilenet v2 also and it seemed to perform well in some cases like this.

![image](https://user-images.githubusercontent.com/61727245/156106932-0acb9866-9cca-4a3c-a1b9-6bfa9bdb928b.png)

For this image, I tested it on Mobilenet v2 and it gave me a higher probability of being porn than inception v3.

![image](https://user-images.githubusercontent.com/61727245/156107096-69731b75-30be-46a9-a30a-9575d99da191.png)

I tested the following image also on Mobilenet v2 and it is giving a false positive.

![image](https://user-images.githubusercontent.com/61727245/156107761-04efcf3a-829e-4b12-beb6-5c28f1337835.png)

This seems to be happening with all the models in the dropdown.

**Comment by GantMan** [comment]
This is good to know.  I'll add these to the next training set.

**Comment by adnankhan37** [comment]
Since I am actively using the Inception v3 model in my project, may I know when will I be able to download the updated Inception v3 model trained on these images? :)

Thanks!

**Comment by GantMan** [comment]
Did you say that one of the other models was better at not being tricked?   Because I'm going to add a voting classifier hat will help.

**Comment by adnankhan37** [comment]
Yes, that would be Mobilenet v2, but just in a few cases, it performed better than Inception v3. 
But when Mobilenet v2 gave us a False Positive for an image, the probability it gave was higher than the one given by the Inception v3 model.

**Comment by adnankhan37** [comment]
Hi,

I might have an idea of what is going wrong with the coffee/tea cup images.
I found the following repository of the NSFW dataset with girls holding teacups which you have mentioned in the readme file.

Link: https://github.com/EBazarov/nsfw_data_source_urls/tree/master/raw_data/appearance/reddit_sub_TeaGirls

Can that be a possible reason for us getting False Positives on the cup images?

Thanks!

**Comment by GantMan** [comment]
Ahhhh, that makes sense.   Strange!  Yeah, if it would be important to counter-balance this kind of data so the AI doesn't form a bias against teacups.

**Comment by adnankhan37** [comment]
Yes, that sounds great. It would be quite helpful if the model is trained on teacup images and then we can use that updated model in our project.

Thanks!

**Comment by adnankhan37** [comment]
Hi, is there any update on the training? 
I found more false positives on the following images. I tested them on Inception v3.
Thanks!

<img width="310" alt="Screenshot 2022-03-07 164702" src="https://user-images.githubusercontent.com/61727245/158166154-fe3fa713-9bd9-4dcd-b833-54f1f3dbe464.png">

<img width="331" alt="Screenshot 2022-03-07 163152" src="https://user-images.githubusercontent.com/61727245/158166313-77fda0bd-c26b-4483-9dcb-678b84216ee0.png">

<img width="388" alt="Screenshot 2022-03-07 162333" src="https://user-images.githubusercontent.com/61727245/158166589-cf786e83-07b9-4563-a0d5-d158ffb3ffe7.png">

**Comment by GantMan** [comment]
I've got a few other tasks before retraining this model on the new images.   My goal will be to create some metrics for retraining.  Unfortunately those are behind ANOTHER item I need to complete.   If this is a blocker I highly recommend you don't wait for me, and perform transfer learning to fix any bias you find.

Since this is a free open source project, the progress is relegated to my free time for OSS.
